packages/components/src/presentation-components/properties/UniquePropertyValuesTargetSelector.tsx

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useState } from "react";
+import { ActionMeta, MultiValue } from "react-select";
+import { PropertyDescription, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { PropertyFilterBuilderRuleValueProps } from "@itwin/components-react";
+import { IModelConnection } from "@itwin/core-frontend";
+import { ContentSpecificationTypes, Descriptor, DisplayValueGroup, FieldDescriptor, KeySet, Ruleset, RuleTypes, Value } from "@itwin/presentation-common";
+import { Presentation } from "@itwin/presentation-frontend";
+import { AsyncMultiTagSelect } from "../instance-filter-builder/MultiTagSelect";
+import { PRESENTATION_TARGETS_BATCH_SIZE } from "./UseNavigationPropertyTargetsLoader";
+import { translate } from "../common/Utils";
+
+/** @internal */
+export function UniquePropertyValuesTargetSelector(props: PropertyFilterBuilderRuleValueProps & { imodel: IModelConnection; descriptor: Descriptor }) {
```

Comment:
I don't think `target` in this this component makes sense. I would suggest removing `target` from functions and variables names.

```suggestion
export function UniquePropertyValuesSelector(props: PropertyFilterBuilderRuleValueProps & { imodel: IModelConnection; descriptor: Descriptor }) {
```

Reply:
fixed